upgrade/1.4.4/README.md

```diff
@@ -0,0 +1,392 @@
+1. [Enable `smartmon` metrics on storage NCNs](#enable-smartmon-metrics-on-storage-ncns)
+1. [Update management node CFS configuration](#update-management-node-cfs-configuration)
+1. [Update NCN images](#update-ncn-images)
+    1. [Warnings](#warnings)   
+    1. [Image customization](#image-customization)
+1. [Storage nodes in-place update](#storage-nodes-in-place-update)   
+1. [Kubernetes nodes rolling rebuild](#kubernetes-nodes-rolling-rebuild)   
+1. [Update test suite packages](#update-test-suite-packages)
+1. [Verification](#verification)
+1. [Take Etcd manual backup](#take-etcd-manual-backup)
+1. [Complete upgrade](#complete-upgrade)
+
+### Preparation
+
+1. (`ncn-m001#`) Start a typescript on `ncn-m001` to capture the commands and output from this procedure.
+
+   ```bash
+   script -af csm-update.$(date +%Y-%m-%d).txt
+   export PS1='\u@\H \D{%Y-%m-%d} \t \w # '
+   ```
+
+1. Download and extract the CSM `v1.4.4` release to `ncn-m001`.
+
+   See [Download and Extract CSM Product Release](../../update_product_stream/README.md#download-and-extract-csm-product-release).
+
+1. (`ncn-m001#`) Set `CSM_DISTDIR` to the directory of the extracted files.
+
+   ***IMPORTANT*** If necessary, change this command to match the actual location of the extracted files.
+
+   ```bash
+   export CSM_DISTDIR="$(pwd)/csm-1.4.4"
+   echo "${CSM_DISTDIR}"
+   ```
+
+1. (`ncn-m001#`) Set `CSM_RELEASE_VERSION` to the CSM release version.
+
+   ```bash
+   export CSM_RELEASE_VERSION="$(${CSM_DISTDIR}/lib/version.sh --version)"
+   echo "${CSM_RELEASE_VERSION}"
+   ```
+
+1. Download and install/upgrade the **latest** documentation on `ncn-m001`.
+
+   See [Check for Latest Documentation](../../update_product_stream/README.md#check-for-latest-documentation).
+
+### Setup Nexus
+
+(`ncn-m001#`) Run `lib/setup-nexus.sh` to configure Nexus and upload new CSM RPM repositories, container images, and
+Helm charts:
+
+```bash
+cd "$CSM_DISTDIR" && ./lib/setup-nexus.sh
+echo "RC=$?"
+```
+
+On success, the output should end with the following:
+
+```text
++ Nexus setup complete
+setup-nexus.sh: OK
+RC=0
+```
+
+In the event of an error,
+consult [Troubleshoot Nexus](../../operations/package_repository_management/Troubleshoot_Nexus.md)
+to resolve potential problems and then try running `setup-nexus.sh` again. Note that subsequent runs of `setup-nexus.sh`
+may report `FAIL` when uploading duplicate assets. This is okay as long as `setup-nexus.sh` outputs `setup-nexus.sh: OK`
+and exits with status code `0`.
+
+### Update Argo CRDs
+
+(`ncn-m001#`) Run the following script in preparation for 1.4.4 patch upgrade:
+
+```bash
+for c in $(kubectl get crd | grep argo | cut -d' ' -f1); do
+   kubectl label --overwrite crd $c app.kubernetes.io/managed-by="Helm"
+   kubectl annotate --overwrite crd $c meta.helm.sh/release-name="cray-nls"
+   kubectl annotate --overwrite crd $c meta.helm.sh/release-namespace="argo"
+done
```

Comment:
Seems like we should check if the commands passed or not.

```suggestion
function run_cmd {
  "$@" && return 0 || echo "ERROR: Command failed with rc $?: $*" >&2 ; return 1
}

for c in $(kubectl get crd | grep argo | cut -d' ' -f1); do
   run_cmd kubectl label --overwrite crd $c app.kubernetes.io/managed-by="Helm" || break
   run_cmd kubectl annotate --overwrite crd $c meta.helm.sh/release-name="cray-nls" || break
   run_cmd kubectl annotate --overwrite crd $c meta.helm.sh/release-namespace="argo" || break
done
```

Reply:
I blame 1.4.3, since I copied it from that :)